kustomize/commands/edit/add/addreplacement_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	testutils_test "sigs.k8s.io/kustomize/kustomize/v5/commands/internal/testutils"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const (
+	replaceFileName    = "replacement.yaml"
+	replaceFileContent = `this is just a test file`
+)
+
+func TestAddReplacementWithFilePath(t *testing.T) {
+	fSys := filesys.MakeEmptyDirInMemory()
+	err := fSys.WriteFile(replaceFileName, []byte(replaceFileContent))
+	require.NoError(t, err)
+	testutils_test.WriteTestKustomization(fSys)
+
+	cmd := newCmdAddReplacement(fSys)
+	args := []string{
+		"--path", patchFileName,
+	}
+	cmd.SetArgs(args)
+	assert.NoError(t, cmd.Execute())
+	content, err := testutils_test.ReadTestKustomization(fSys)
+	assert.NoError(t, err)
+	for i := 1; i < len(args); i += 2 {
+		assert.Contains(t, string(content), args[i])
+	}
```

Comment:
Is there a more meaningful assertion that could be done here?
I'm concerned this test might pass by happenstance without verifying that the result was what we expected to happen.

A test similar to this would be more reliable:
```go
kf, err := kustfile.NewKustomizationFile(fSys)
require.NoError(t, err)

kustomization, err := kf.Read()
require.NoError(t, err)

for _, replacement := range kustomization.Replacements {
    require.Equal(t, expectedPath, replacement.Path)
}
```

Reply:
update as suggested

Reply:
Thanks for updating this!